src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -285,5 +378,31 @@ public override unsafe bool TryEncodeUnicodeScalar(int unicodeScalar, char* buff
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool NeedsEscaping(char value) => value > LastAsciiCharacter || AllowList[value] == 0;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int NeedsEscaping(Vector128<sbyte> sourceValue)
+        {
+            // Check if any of the 16 bytes need to be escaped.
+            Vector128<sbyte> mask = Ssse3.IsSupported
+                ? Ssse3Helper.CreateEscapingMask_DefaultJavaScriptEncoderBasicLatin(sourceValue)
+                : Sse2Helper.CreateEscapingMask_DefaultJavaScriptEncoderBasicLatin(sourceValue);
```

Comment:
I haven't looked at the rest of the code changes yet, but don't we need an `Sse2.IsSupported` check as well?

Reply:
This method is only called from places where `Sse2.IsSupported` is already checked. So if SSSE3 is not supported, it is ensured that SSE2 is available. I'll add a `Debug.Assert` to make this more obvious.